TrafficCapture/nettyWireLogging/src/main/java/org/opensearch/migrations/trafficcapture/netty/LoggingHttpResponseHandler.java

```diff
@@ -102,6 +104,12 @@ public void deregister(ChannelHandlerContext ctx, ChannelPromise promise) throws
 
     @Override
     public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise) throws Exception {
+        // Enable responses for blocking requests to have the same blocking metadata
+        Attribute<Object> attr = ctx.channel().attr(AttributeKey.valueOf("isBlocking"));
+        if (attr.get() != null) {
+            trafficOffloader.setIsBlockingMetadata(true);
+            attr.set(null);
```

Comment:
I don't understand what is happening here.  What if the attribute was set to "false"?  Why are you setting the blocking flag once, but then unsetting it after that?  It seems like this conditional should never be in here.  Instead, when the metadata that you need is found, it should be sent to the serializer and that should have the responsibility to track it.  

As it is now, I'm presuming that the request's offloading no longer needs to block.  While that is true, that behavior 1) should not be decided here, nor is it.  2) You're not passing the "isBlocking" flag to indicate or control blocking behavior, but to route messages to the right topic and splitting a request/response pair between multiple topics is incorrect.

Reply:
This section and the other section for attribute were more of a hack to get things in place for supporting multiple topics. I've removed them now and just let the serializer dictate this in a similar way. However, this piece is still not complete. I feel we need to add some mechanism for linking request and responses together through some UUID or other mechanism. Currently there doesn't seem to be a reliable way to tell what response is for which request. Let me know if you have thoughts on this.